### PR TITLE
fix: Alpha should not be able to edit datasets that they don't own

### PR DESCRIPTION
**Pull request by hughhhh**
### SUMMARY

There is no longer a config value for `OLD_API_CHECK_DATASET_OWNERSHIP` due to 2.0 release. This is now allowing all users who have write permissions to edit any dataset. Adding this logic will verify if this user is an admin or owner before completing this request.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

https://user-images.githubusercontent.com/27827808/165382733-f4402f07-232c-4fb8-b445-4106302eb75f.mov

Screen Shot 2022-04-26 at 8 50 12 PM

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by jess-dillard** [comment]
@hughhhh Looking good! Only thing I'm seeing is the disabled button shouldn't have a hover state (it should remain grey even when the user hovers)

**Comment by hughhhh** [comment]
Screen Shot 2022-04-26 at 3 14 17 PM

File: superset-frontend/src/views/CRUD/data/dataset/DatasetList.tsx [hunk 1]
```diff
@@ -344,6 +365,11 @@ const DatasetList: FunctionComponent<DatasetListProps> = ({
       },
       {
         Cell: ({ row: { original } }: any) => {
+          // Verify owner or isAdmin
+          const allowEdit =
+            original.owners.map((o: any) => o.id).includes(user.userId) ||
```
Comment (hughhhh) [inline superset-frontend/src/views/CRUD/data/dataset/DatasetList.tsx:370]: add `Owners` to for loop

**Review by pkdotson — APPROVED** [review]
code looks good to me.

**Comment by yousoph** [comment]
Ephemeral looks good to me (and design + QA took a look as well) :)

**Comment by jinghua-qa** [comment]
tested in the ephermal env, primary contributor can not edit/delete the dataset they did not own. LGTM!

**Comment by sadpandajoe** [comment]
🏷️ preset:2022.17

**Pull request merged**